### PR TITLE
updated patch behaviour for azureHybridBenefit property in connectedk8s

### PR DESCRIPTION
**Pull request by akashkeshari**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
connectedk8s

**Review by sirireddy12 — APPROVED** [review]

File: src/connectedk8s/azext_connectedk8s/custom.py [hunk 1]
```diff
@@ -947,6 +947,12 @@ def update_connected_cluster(cmd, client, resource_group_name, cluster_name, htt
         patch_cc_response = update_connected_cluster_internal(client, resource_group_name, cluster_name, tags, distribution, distribution_version, azure_hybrid_benefit)
 
     proxy_params_unset = (https_proxy == "" and http_proxy == "" and no_proxy == "" and proxy_cert == "" and not disable_proxy)
+
+    # Returning the ARM response if only AHB is being updated
+    arm_properties_only_ahb_set = (tags is None and distribution is None and distribution_version is None and azure_hybrid_benefit is not None)
+    if proxy_params_unset and auto_upgrade is None and container_log_path is None and arm_properties_only_ahb_set:
+        return patch_cc_response
```
Comment (zhoxing-ms) [inline src/connectedk8s/azext_connectedk8s/custom.py:954]: Could you add a test case for this change?

Comment (akashkeshari) [reply src/connectedk8s/azext_connectedk8s/custom.py:954]: @zhoxing-ms , The test case for that is already covered in the test file
https://github.com/Azure/azure-cli-extensions/blob/1c98cce60f103a48ca1bb8818208bfd7899155b0/src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py#L50

Comment (zhoxing-ms) [reply src/connectedk8s/azext_connectedk8s/custom.py:954]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**